tests/components/homekit_controller/common.py

```diff
@@ -174,19 +168,8 @@ def add_characteristic(self, name):
         }
     }
 
-    fire_service_discovered(hass, SERVICE_HOMEKIT, discovery_info)
-    await hass.async_block_till_done()
-
-    # We can't guarantee what our entity_id will be so we have to find it
-    # in hass after its been created. (The cover entity in particular unsets
-    # its name, meaning its unique_id is used and that isn't stable enough to
-    # rely on)
-    entities = list(hass.data[domain].entities)
-
-    # Right now we only support testing a single entity at a time
-    # This will need extending to find the new entity in the list if more
-    # entities are needed at once
-    assert len(entities) == 1
-    entity = entities[0]
+    with mock.patch.object(HomeKitEntity, 'name', 'testdevice'):
```

Comment:
If we can, we should avoid patching the entity, since that's what we're testing. Can we patch the source that the entity is using instead?

Reply:
Not really - there is a conflict between the non-determinism of the upstream homekit models and the way some (but not all) of these entities are written.

If possible i'd like to keep it this way and remove it again completely with the PR i mentioned above.

Reply:
Ok.